common-npm-packages/azure-arm-rest-v2/azure-arm-common.ts

```diff
@@ -149,16 +152,29 @@ export class ApplicationTokenCredentials {
         return this.clientId;
     }
 
-    public getToken(force?: boolean): Q.Promise<string> {
-        return this.useMSAL ? this.getMSALToken(force) : this.getADALToken(force);
+    public async getToken(force?: boolean): Promise<string> {
+        // run exclusively to prevent race conditions
+        const release = await this.tokenMutex.acquire();
+
+        try {
+            let promisedTokenResult = this.useMSAL ? this.getMSALToken(force) : this.getADALToken(force);
```

Comment:
Lets use `const` as this variable is not reassigned.

Reply:
It is updated.